web/src/server_events_dispatch.js

```diff
@@ -646,13 +645,11 @@ export function dispatch_normal_event(event) {
                     break;
                 case "delete":
                     for (const stream_id of event.stream_ids) {
-                        const sub = sub_store.get(stream_id);
-                        const is_subscribed = sub.subscribed;
+                        const was_subscribed = sub_store.get(stream_id).subscribed;
                         const is_narrowed_to_stream = narrow_state.narrowed_to_stream_id(stream_id);
                         stream_data.delete_sub(stream_id);
-                        stream_settings_ui.update_settings_for_archived(sub);
-                        message_view_header.maybe_rerender_title_area_for_stream(stream_id);
```

Comment:
We did not rerender the title and message view before the changes made for archiving streams. So have done that for now, but am not sure what would be the ideal behavior.

This is how the topic narrowed view is live updated if title and message view is not re-rendered -
![image](https://github.com/user-attachments/assets/42103b0b-79fb-458b-88e6-956816299153)

This is how the UI is live updated when rerendering -
![image](https://github.com/user-attachments/assets/04e87c94-0eec-4835-aaa9-f4bb1b75dba4)

Reply:
When reloading the same page, the header is shown as "Unknown channel" as in the second screenshot and messages are also not shown with text mentioning that channel does not exist.

Reply:
To me that looks like a bug in the rendering code -- a topic view in an archived channel should never shown "unknown channel" if the client still has metadata access to the channel.

(Ideally, we'd show the channel name with the archived icon).

Reply:
I was not clear before. I am talking about the behavior when user loses access to stream.